### PR TITLE
Dist block

### PR DESCRIPTION
**Pull request by phermes**
Not to merge!

As discussed in #111 we want a new routine to sample an initial distribution. 

Thus far, we can read a DIST block in fort.3, load an initial file with the physical distribution. Also, the emittance can be defined in the DIST block, to allow for reading a normalized distribution. Next step is a subroutine to de-normalize the coordinates. 

The changes come along with the definition of new arrays required to load and track heavy-ions:

naa(j) = nuclear mass number
nzz(j) = nuclear charge number
nucm(j) = nuclear mass (in GeV/c^2)
zz0,aa0,nucm0 correspond to the reference particle 
mtc(j)  = mass to charge relative to reference particle = (nzz(j)*nucm0)/(zz0*nucm(j))
moidpsv(j) = mtc(j)*oidpsv(j)
omoidpsv(j) = c1e3*((one-mtc(j))*oidpsv(j)) 

so far, however, the heavy-ion properties are ignored. This will require modifications of the tracking maps. We can, however sample the heavy-ion properties already now. 

```
DISTRIBUTION
READ        initial.dat NORMALIZED
EMITTANCE   1           3.5e-6
EMIT        2           3.1e-6
EMITA       3           2.2e-4
/NUCLI     193.6876520557960   208             82
/DTYPE     1 FLATC   5.500000D+00    5.550000D+00
/DTYPE     2 GAUSS
/DTYPE     3 GAUSS
NEXT
HION
82 208 193.6876520557960
NEXT
```

Also, there should be a veto at all places where a distribution is read from another block. I put a veto to the collimation routine already. Where else should we put it?

### CODE REVIEW

File: SixTrack/sixtrack.s [hunk 1]
```diff
@@ -2036,13 +2062,13 @@ C     Block with data/fields needed for checkpoint/restart of DYNK
 +cd kickvxxh
 +if .not.tilt
             yv(1,j)=yv(1,j)+(strack(i)*oidpsv(j))*crkve                  !hr02
-            yv(2,j)=yv(2,j)-(strack(i)*oidpsv(j))*cikve                  !hr02
+            yv(2,j)=yv(2,j)-(strack(i)*oidpsv(j))*cikve !hr02
```
Comment (kyrsjo) [inline SixTrack/sixtrack.s]: Please do not modify spacing unnecessarily

File: SixTrack/sixtrack.s [hunk 2]
```diff
@@ -12274,6 +12308,8 @@ cc2008
       if(idat.eq.dynk)  goto 2200
       if(idat.eq.fma)   goto 2300
       if(idat.eq.elens) goto 2400
+      if(idat.eq.dist)  goto 2500
```
Comment (kyrsjo) [inline SixTrack/sixtrack.s]: Wire (Pull request #174, likely to be merged before DIST) already uses `goto 2500`, could you use 2600 and 2700?

File: SixTrack/sixtrack.s [hunk 3]
```diff
@@ -62993,6 +63435,270 @@ c$$$     &           myalphay * cos(phiy))
...
+      enddo
+!     initialise particle counter
+      jj = 0      
+
+
+      write(*,*), '--> DIST, dist_read_unit', dist_read_unit
+      write(*,*), '--> DIST, npart         ', npart
+      write(*,*), '--> DIST, napx          ', napx
+      
+      open( unit=dist_read_unit, file=dist_fname )
```
Comment (kyrsjo) [inline SixTrack/sixtrack.s]: To demand that the file should already exist, add `state=OLD`

Also, check that the unit number is not taken by `inquire`

**Comment by kyrsjo** [comment]
Hi, there has been a bunch of changes lately, especially to the build system. It might be a good idea for you to merge master into your branch, in order to keep things synchronized.

**Comment by kyrsjo** [comment]
Hi again,
I just merged #233 into the master branch, removing all the duplication of write statements. You may want to (1) merge this to your branch and (2) do the same to your branch, i.e. always use lout for writes, never `(*,*)`.

Also, please consider not committing extra files like `SixTrack/sixtrack_BASE_26516.s` etc. to the repository - it will show up in the history of the master branch (together with a lot of other junk, but still)...

Best regards,
Kyrre

**Comment by rdemaria** [comment]
closing because the interfaces are all changed, development should continue in the issue #111

**Pull request closed without merging**